qiskit/providers/__init__.py

```diff
@@ -110,6 +109,381 @@
+execute circuits. It also defines the interface for a backend to provide the
+necessary information to describe the backend to the compiler so that it can
+embed and optimize any circuit to the backend. There are 3 required things
+in every backend object, a backend configuration object set in the
+`__init__()`, a ``run()`` method, and a ``_default_options()`` method. For
+example, a minimum working example would be something like::
+
+    from qiskit.providers import BackendV1 as Backend
+    from qiskit.providers.models import BackendConfiguration
+    from qiskit.providers import Options
+
+
+    class MVbackend(Backend):
+
+        configuration = {
+            'backend_name': 'my_backend',
+            'backend_version': '1.0.0',
+            'n_qubits': 5,
+            'basis_gates': ['p', 'sx', 'u', 'cx', 'id'],
+            'coupling_map': [[0, 1], [1, 2], [2, 3], [3. 4]],
+            'simulator': False,
+            'local': False,
+            'conditional': True,
+            'open_pulse': False,
+            'memory': True,
+            'max_shots': 8196,
+            'gates': []
+        }
+
+        def __init(self, provider):
+            super().__init__(
+                configuration=BackendConfiguration.from_dict(self.configuration),
+                provider=provider)
+
+        @classmethod
+        def _default_options(cls):
+            return Options(shots=1024, memory=False)
+
+        def run(circuits, **kwargs):
+            # serialize circuits submit to backend and create a job
+            for kwarg in kwargs:
+                if not hasattr(kwarg, self.options):
+                    warnings.warn(
+                        "Option %s is not used by this backend" % kwarg,
+                        UserWarning, stacklevel=2)
+            options = {
+                'shots': kwargs.get('shots', self.options.shots)
+                'memory': kwargs.get('memory', self.options.shots),
+            }
+            job_json = convert_to_wire_format(circuit, options)
+            job_handle = submit_to_backend(job_jsonb)
+            return MyJob(self. job_handle, job_json, circuit)
+
+
+Transpiler Interface
+--------------------
+
+The key piece of the backend object is how it describes itself to the compiler.
+This is composed of three classes, the
+:class:`~qiskit.providers.models.BackendConfiguration` which describes the
+immutable properties of the backend (things like the number of qubits, coupling
+map, etc), the :class:`~qiskit.providers.models.BackendProperties` which
+described measured properties of the device (such as current error rates), and
+the :class:`~qiskit.providers.models.PulseDefaults` which defines the
+default pulse behavior for the backend (such as the pulse sequences for gates).
+Of these the only required piece for a backend is the
+:class:`~qiskit.providers.models.BackendConfiguration`, the other information
+is only used if the backend either returns the necessary information.
```

Comment:
This doesn't make sense.  Do you mean that the other information is only used if the backend _fails_ to return the necessary information?  Also, since you've been describing the backend classes, you can say "Of these only :class:`~qiskit.providers.models.BackendConfiguration` is required" or something like that.  Same meaning, but more concise.

Reply:
Fixed in: https://github.com/Qiskit/qiskit-terra/pull/5406/commits/922ae3cc737a960f608e297abdcda8d16f0fb0b0 (I also dropped the second half about the optional pieces it was redundant and confusing as you pointed out.